### PR TITLE
Respect group id in options for ranking task

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
Fix https://github.com/dotnet/machinelearning/issues/2652

### CODE REVIEW

File: test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs [hunk 1]
```diff
@@ -115,7 +115,8 @@ public void FastTreeRankerEstimator()
                 new FastTreeRankingTrainer.Options
                 {
                     FeatureColumn = "NumericFeatures",
-                    NumTrees = 10
+                    NumTrees = 10,
+                    GroupIdColumn = "Group"
```
Comment (singlis) [inline test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs]: >GroupIdColumn [](start = 20, length = 13)

I feel like this should be "GroupColumn" to fall more inline with FeatureColumn and LabelColumn.

File: src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs [hunk 2]
```diff
@@ -163,7 +164,7 @@ private protected virtual void CheckDataValid(IChannel ch, RoleMappedData data)
             ch.CheckParam(data.Schema.Label.HasValue, nameof(data), "Need a label column");
         }
 
-        protected virtual void GetDefaultParameters(IChannel ch, int numRow, bool hasCategarical, int totalCats, bool hiddenMsg=false)
+        protected virtual void GetDefaultParameters(IChannel ch, int numRow, bool hasCategarical, int totalCats, bool hiddenMsg = false)
```
Comment (singlis) [inline src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs]: >hasCategarical [](start = 82, length = 14)

misspelling -- should be hasCategorical.

Comment (singlis) [inline src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs]: >hiddenMsg [](start = 118, length = 9)

for the sake of not having short names -hiddenMessage?

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs]: I'll leave that stuff to whoever would scrub lightgbm learners.

**Pull request merged**